pkg/transportrequest/uploadcts_test.go

```diff
@@ -0,0 +1,156 @@
+			},
+			TransportRequestID: "12345678",
+			ConfigFile:         "ui5-deploy.yaml",
+			DeployUser:         "node",
+		}
+		err := action.Perform(&cmd)
+		if assert.NoError(t, err) {
+			assert.Contains(
+				t,
+				cmd.Calls[0],
+				"npm install --global --verbose --registry https://registry.example.org @sap/my-dep",
+			)
+			assert.Contains(
+				t,
+				cmd.Calls[0],
+				"su node",
+			)
+		}
+	})
+
+	t.Run("deploy command tests", func(t *testing.T) {
+		t.Run("all possible values provided", func(t *testing.T) {
+			cmd := mock.ShellMockRunner{}
+			action := CTSUploadAction{
+				Connection:  CTSConnection{Endpoint: "https://example.org:8080/cts", Client: "001", User: "me", Password: "******"},
+				Application: CTSApplication{Pack: "abapPackage", Name: "appName", Desc: "the Desc"},
+				Node: CTSNode{
+					DeployDependencies: []string{},
+					InstallOpts:        []string{},
+				},
+				TransportRequestID: "12345678",
+				ConfigFile:         "ui5-deploy.yaml",
+				DeployUser:         "doesNotMatterInThisCase",
+			}
+
+			err := action.Perform(&cmd)
+			if assert.NoError(t, err) {
+				assert.Contains(
+					t,
+					cmd.Calls[0],
+					"fiori deploy -f -y --username ABAP_USER --password ABAP_PASSWORD -e \"the Desc\" --noConfig -u https://example.org:8080/cts -l 001 -t 12345678 -p abapPackage -n appName",
+				)
+				assert.Equal(t, []string{"ABAP_USER=me", "ABAP_PASSWORD=******"}, cmd.Env)
+			}
+		})
+
+		t.Run("all possible values omitted", func(t *testing.T) {
+			// In this case the values are expected inside the fiori deploy config file
+			cmd := mock.ShellMockRunner{}
+			action := CTSUploadAction{
+				Connection:  CTSConnection{Endpoint: "", Client: "", User: "me", Password: "******"},
+				Application: CTSApplication{Pack: "", Name: "", Desc: ""},
+				Node: CTSNode{
+					DeployDependencies: []string{},
+					InstallOpts:        []string{},
+				},
+				TransportRequestID: "12345678",
+				ConfigFile:         "ui5-deploy.yaml",
+				DeployUser:         "doesNotMatterInThisCase",
+			}
+			err := action.Perform(&cmd)
+			if assert.NoError(t, err) {
+				assert.Contains(
+					t,
+					cmd.Calls[0],
+					"fiori deploy -f -y --username ABAP_USER --password ABAP_PASSWORD -e \"Deployed with Piper based on SAP Fiori tools\" --noConfig -t 12345678",
+				)
+				assert.Equal(t, []string{"ABAP_USER=me", "ABAP_PASSWORD=******"}, cmd.Env)
+			}
+		})
+	})
+
+	t.Run("config file releated tests", func(t *testing.T) {
+		connection := CTSConnection{Endpoint: "", Client: "", User: "me", Password: "******"}
+		app := CTSApplication{Pack: "", Name: "", Desc: ""}
+		node := CTSNode{
+			DeployDependencies: []string{},
+			InstallOpts:        []string{},
+		}
+		t.Run("default config file exists", func(t *testing.T) {
+			filesMock := mock.FilesMock{}
+			filesMock.AddFile("ui5-deploy.yaml", []byte{})
+			files = &filesMock
+			defer func() { files = piperutils.Files{} }()
```

Comment:
Why define the filesMock twice? Is the outer filesMock instance necessary?

Reply:
I created the outer mock in order to decouple from the real file system of the machine running the tests. So the outer mock is IMO required in order to avoid test failures since some not expected files are present in that file system. But I changed the code so that we reset each time to the outer file system mock rather than resetting to a non-mock.